recipes/ccache/all/conanfile.py

```diff
@@ -67,23 +53,19 @@ def requirements(self):
             self.requires("hiredis/1.1.0")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
+        if self.settings.compiler.cppstd:
             check_min_cppstd(self, self._min_cppstd)
-
-        def lazy_lt_semver(v1, v2):
-            lv1 = [int(v) for v in v1.split(".")]
-            lv2 = [int(v) for v in v2.split(".")]
-            min_length = min(len(lv1), len(lv2))
-            return lv1[:min_length] < lv2[:min_length]
-
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version and lazy_lt_semver(str(self.settings.compiler.version), minimum_version):
-            raise ConanInvalidConfiguration(
-                "{} requires C++{}, which your compiler does not support.".format(self.name, self._min_cppstd)
-            )
-
-    def package_id(self):
-        del self.info.settings.compiler
+        check_min_vs(self, 192)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+        if Version(conan_version).major < 2 and \
```

Comment:
This is not related to the Conan version. Those docker images are available for any Conan client version, including 1.x, so in case someone tries to use `docker.io/conanio/clang11-ubuntu16.04:1.59.0` will suffer the same error. I need to investigate it further.